File: 5. Longest Palindromic Substring/5. Longest Palindromic Substring.go
func longestPalindrome(s string) string {
    n := len(s)
    if n < 2 {
        return s
    }

    start, maxLength := 0, 1

    for i := 0; i < n; i++ {
        // Odd length palindromes
        left, right := i, i
        for left >= 0 && right < n && s[left] == s[right] {
            if right-left+1 > maxLength {
                start = left
                maxLength = right - left + 1
            }
            left--
            right++
        }

        // Even length palindromes
        left, right = i, i+1
        for left >= 0 && right < n && s[left] == s[right] {
            if right-left+1 > maxLength {
                start = left
                maxLength = right - left + 1
            }
            left--
            right++
        }
    }

    return s[start : start+maxLength]
}

File: 5. Longest Palindromic Substring/5. Longest Palindromic Substring.rs
impl Solution {
    pub fn longest_palindrome(s: String) -> String {
        let n = s.len();
        if n < 2 {
            return s;
        }

        let s_bytes = s.as_bytes();
        let (mut start, mut max_length) = (0, 1);

        for i in 0..n {
            // Odd length palindromes
            let (mut left, mut right) = (i, i);
            while left < n && right < n && s_bytes[left] == s_bytes[right] {
                if right - left + 1 > max_length {
                    start = left;
                    max_length = right - left + 1;
                }
                if left == 0 || right == n - 1 {
                    break;
                }
                left -= 1;
                right += 1;
            }

            // Even length palindromes
            let (mut left, mut right) = (i, i + 1);
            while left < n && right < n && s_bytes[left] == s_bytes[right] {
                if right - left + 1 > max_length {
                    start = left;
                    max_length = right - left + 1;
                }
                if left == 0 || right == n - 1 {
                    break;
                }
                left -= 1;
                right += 1;
            }
        }

        s[start..start + max_length].to_string()
    }
}